### PR TITLE
feat(usenet): add Pesto season uploads

### PR DESCRIPTION
**Pull request by cyb3rRunn3rr**
Hey, this adds proper Pesto season uploads to UA.

The basic idea is pretty simple: Pesto posts every ep on its own, writes one NZB per ep, then builds the full season NZB at the end. UA can send all of those to the selected Usenet indexers.

### What's in here

- new `pesto_season_upload` config switch
- Pesto gets `--season` for TV season packs
- individual ep NZBs + the final season pack are collected and submitted
- new `--usenet-episodes-only INDEXER[,INDEXER...]` arg
  - useful when the pack is already on one indexer and you only wanna add the eps there
  - other selected indexers can still get both the eps and the pack
- the arg is available in the WebUI arguments list too
- dupe checks run per ep, not just against the full pack
- each ep gets its own UUID/cache, MediaInfo and screenshots
- Curupira gets the configured screenshot count from the actual ep being uploaded
- clearer output showing uploaded/dupe/skipped/failed status for every ep
- existing complete NZB batches are reused, so an indexer retry doesn't repost the whole thing to NNTP
- sidecars are filtered and weird top-level entries like samples/extras fail early instead of becoming their own release
- Pesto temp/config/NZB paths now use UA's writable staging area, which also fixes read-only Docker homes
- keeps the managed 7-Zip setup working, including Windows DLLs

This is Pesto-only. Nyuu behavior stays as-is.

### Checks

- tested with a real season upload through the WebUI
- ep NZBs and the season NZB were created correctly
- episodes-only routing and per-ep screenshots tested against Curupira
- `1641 passed, 3 skipped`
- Ruff + formatting clean
- Docker builds passed on amd64 and arm64

## Summary by CodeRabbit

- **New Features**
  - Added Pesto season uploads for TV packs, uploading episodes individually alongside a consolidated season NZB.
  - Added `--usenet-episodes-only` to send episode NZBs—but not the final season pack—to selected indexers.
  - Added configuration and Web UI support for season uploads and episode-only submissions.
  - Added configurable Pesto filename obfuscation modes, including light and full obfuscation.

- **Documentation**
  - Documented Pesto season upload behavior, configuration, supported scenarios, obfuscation modes, and indexer selection.

### CODE REVIEW

**Comment by cyb3rRunn3rr** [comment]
hey wastaken,

tested this here on a few different indexers and everything worked fine so far, no issues on my side

I also think the `--usenet-episodes-only` flag is worth having. sometimes an indexer already has the full season pack but is still missing the individual eps

Curupira is one example where I ran into this. in cases like that I just wanna send the missing eps without uploading the same pack again, while still letting the other selected indexers get both if needed

CodeRabbit also caught an edge case where Pesto could create the episode NZBs but fail to create the final season pack. that one slipped past me since the original test always had the pack being created

fixed it now. the eps are still kept and submitted, none of them gets mistaken for the pack, and a retry can reuse the valid episode NZBs instead of posting them all again. also added tests for both cases, with and without the final pack

ran everything again after the fix, 1644 passed and 3 skipped

if you think anything else should be changed or handled differently, just let me know and I’ll adjust it

cheers

**Comment by cyb3rRunn3rr** [comment]
@coderabbitai review

**Comment by cyb3rRunn3rr** [comment]
hey wastaken,

one more thing came up during the final real-world test

the individual Pesto eps were doing the normal broad dupe check on Usenet, so an existing ep from another release group could make the new one get skipped. since these generated ep uploads run unattended after the main approval, there was no prompt to upload anyway either

fixed that now. for Pesto season eps on Usenet, it only blocks the upload when the exact same release is already there. another group, or an existing full season pack, won't block the individual ep anymore

this is scoped only to the generated Pesto episode submissions on Usenet. normal tracker dupe behavior and the final season pack behavior are unchanged

also fixed the Newznab result parsing so a release title isn't treated like a fake file list during exact matching

tested the exact same release vs different group case, checked the episode/pack classification again, and built the updated Docker image for both amd64 and arm64 without issues

should be ready for review now. if you want anything changed just let me know

cheers

File: src/usenetcreate.py [hunk 1]
```diff
@@ -901,8 +1097,14 @@ async def prepare_and_upload_usenet(meta: Meta, config: dict[str, Any], *, prepa
     if not input_path:
         logger.error("[red]Error: Input path is missing.[/red]")
         return None
-    import hashlib
-
+    uploader = str(usenet_cfg.get("usenet_uploader", "nyuu")).lower()
+    use_pesto = uploader == "pesto"
+    pesto_obfuscation_mode = str(usenet_cfg.get("pesto_obfuscation_mode", "full")).strip().lower()
+    if pesto_obfuscation_mode not in {"full", "light", "article", "full-shared"}:
+        raise ValueError("USENET.pesto_obfuscation_mode must be one of: full, light, article, full-shared")
```
[the remark this answers is not in the document]
Comment (cyb3rRunn3rr) [reply src/usenetcreate.py]: fixed in bfb7c304. the mode is now validated only when Pesto is the active uploader, so an unused Pesto setting can no longer block Nyuu. kept strict validation for actual Pesto uploads and added coverage for both paths.

File: src/usenetcreate.py [hunk 2]
```diff
@@ -41,6 +42,200 @@ def generate_random_poster() -> str:
...
+            logger.info(f"[red]{tracker_key}: {status['status_message']}[/red]")
+        elif uploaded_episode_counts.get(tracker_key, 0) > 0:
+            status["upload_success"] = True
+            status["status_message"] = f"{uploaded_episode_counts[tracker_key]} episode NZB upload(s) succeeded"
+            if tracker_key in episodes_only_trackers:
+                status["status_message"] += "; season pack skipped by --usenet-episodes-only"
+        elif duplicate_episode_counts.get(tracker_key, 0) > 0 and tracker_key in episodes_only_trackers:
+            status["dupe"] = True
+            status["upload_success"] = False
+            status["status_message"] = "All episode NZBs were duplicates; season pack skipped by --usenet-episodes-only"
```
[the remark this answers is not in the document]
Comment (cyb3rRunn3rr) [reply src/usenetcreate.py]: fixed in bfb7c304. skipped episode results are counted now, and the parent status is only marked as dupe when there are duplicates with no uploads, failures, or skipped eps. added a mixed duplicate + skipped regression test too.

**Comment by cyb3rRunn3rr** [comment]
quick follow-up from another multi-season test:

fixed the two new review points in `bfb7c304`

- an unused invalid Pesto obfuscation setting no longer blocks Nyuu uploads. strict validation still applies when Pesto is actually selected
- mixed episode results are tracked properly now. a batch with duplicated + skipped eps is no longer reported as if every ep was a duplicate

also fixed a real multi-folder issue I hit while processing seasons back to back. the screenshot uploader changes the process cwd to the temporary screenshot dir, then episode cleanup removes that dir. the next season could start with `getcwd() failed: No such file or directory`. the cwd is now always restored before cleanup, including error paths

added regression coverage for all three cases and the focused tests/lint are passing. rebuilding the test image now as well
